apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopRightPanel.tsx

```diff
@@ -60,12 +63,44 @@ export function TlaEditorTopRightPanel({
 	)
 }
 
+function useGetFileName() {
+	const editor = useEditor()
+	const msg = useTranslation()
+	const defaultPageName = msg('page-menu.new-page-initial-name')
+
+	const documentName = editor.getDocumentSettings().name
+	if (documentName?.length > 0) return documentName
+
+	const firstPageName = editor.getPages()[0].name
+	if (
+		firstPageName.length > 0 &&
+		!firstPageName.startsWith('Page 1') &&
```

Comment:
We still default to `Page 1` in some cases (think during ensure store usable, possibly other places).